src/pages/GetStarted/CustomWorkspaceTab/StorageType.tsx

```diff
@@ -0,0 +1,184 @@
+        });
+      }
+    }
+  }
+
+  public componentDidUpdate(prevProps: Props): void {
+    if (prevProps.storageType !== this.props.storageType) {
+      const selected = this.props.storageType;
+      this.setState({ selected });
+    }
+  }
+
+  public componentDidMount(): void {
+    const selected = this.props.storageType ? this.props.storageType : this.preferredType;
+    this.setState({ selected });
+  }
+
+  private handleToggle(isOpen: boolean): void {
+    this.setState({ isOpen });
+  }
+
+  private handleSelect(event, selection): void {
+    if (this.props.onChange) {
+      this.props.onChange(selection);
+    }
+    this.setState({
+      selected: selection,
+      isOpen: false
+    });
+  }
+
+  private handleModalToggle(): void {
+    this.setState(({ isModalOpen }) => ({
+      isModalOpen: !isModalOpen
+    }));
+  }
+
+  private getModalContent(): React.ReactNode {
+    const hasAsync = this.storageTypes.some(type => StorageType[type] === StorageType.async);
+    const getAsync = hasAsync ?
+      <Text><span className={styles.experimentalStorageType}> Experimental feature </span><br />
+        <b>Asynchronous Storage </b>
+        is combination of Ephemeral and Persistent storages. It allows for faster I / O and keeps your changes,
+        it does backup the workspace on stop and restores it on start.</Text> : '';
+    const hasPersistent = this.storageTypes.some(type => StorageType[type] === StorageType.persistent);
+    const getPersistent = hasPersistent ?
+      <Text><b>Persistent Storage</b> is slow I/O but persistent.</Text> : '';
+    const hasEphemeral = this.storageTypes.some(type => StorageType[type] === StorageType.ephemeral);
+    const getEphemeral = hasEphemeral ?
+      <Text><b>Ephemeral Storage</b> allows for faster I/O but may have limited
+        storage and is not persistent.</Text> : '';
+    const href = this.props.brandingStore.data.docs.storageTypes;
+
+    return (<TextContent>
+      {getPersistent}
+      {getEphemeral}
+      {getAsync}
+      <Text><a rel="noreferrer" target="_blank" href={href}>Open documentation page</a></Text>
+    </TextContent>);
+  }
+
+  public render(): React.ReactNode {
+    const { isOpen, selected, isModalOpen } = this.state;
+
+    return (
+      <FormGroup
+        label="Storage Type"
+        fieldId="storage-type"
+      >
+        <Select
+          className={styles.storageTypeSelector}
+          aria-label="Storage Type Selector"
+          aria-labelledby="storage-type-selector-id-1"
+          variant={SelectVariant.single}
+          direction={SelectDirection.down}
+          onToggle={_isOpen => this.handleToggle(_isOpen)}
+          onSelect={(event, selection) => this.handleSelect(event, selection)}
+          selections={selected}
+          isOpen={isOpen}
+        >
+          {this.options.map((option, index) => (
+            <SelectOption
+              key={`storage-type-selector-option-${index}`}
```

Comment:
Why don't you use `option.value` as the key?

Reply:
done